src/App.tsx

```diff
@@ -1,34 +1,98 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>();
+  const [selected, setSelected] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState('all');
+
+  useEffect(() => {
+    getTodos().then(setTodos);
+  }, [todos]);
+
+  const filtredTodos = () => {
+    switch (filter) {
+      case 'all':
+        return todos;
+
+      case 'active':
+        return todos?.filter(todo => todo.completed === false);
+
+      case 'completed':
+        return todos?.filter(todo => todo.completed === true);
+
+      default:
+        return todos;
+    }
+  };
+
+  const queryFilter = (todoList: Todo[] | undefined) => {
```

Comment:
instead of writing `| undefined`

just write the initial value as an empty array in `useState`
`  const [todos, setTodos] = useState<Todo[]>([]);`